chainer/functions/connection/shift.py

```diff
@@ -0,0 +1,142 @@
+import chainer
+from chainer.backends import cuda
+from chainer import function_node
+import chainer.functions
+from chainer.utils import type_check
+
+
+def _pair(x):
+    if hasattr(x, '__getitem__'):
+        return x
+    return x, x
+
+
+if cuda.available:
+    cupy = cuda.cupy
+    shift_gpu = cupy.ElementwiseKernel(
```

Comment:
It slows down `import chainer` if this is run in the toplevel code. How about wrapping with a function?